Add a macOS release scripts

simondemeule opened the pull request:
This adds two simple scripts; one for running macdeployqt to package dependencies into the app, and one to run the build with a debug output.

Related to https://github.com/Xmodal/autonomX/issues/177 and #288 

Closes #288

aalex commented on runmacdeployqt:
```diff
@@ -0,0 +1,7 @@
+# before running this, build the release version of the app in Qt Creator / with qmake, then run this to package all libraries / frameworks into the .app executable
+PATH_TO_QT_BIN=/Library/Qt/5.15.0/clang_64/bin
+PATH_TO_RELEASE_DOT_APP=/Users/simondemeule/GitHub/build-autonomx-Desktop_Qt_5_15_0_clang_64bit-Release/autonomx/autonomx.app
+PATH_TO_QML_DIR=/Users/simondemeule/GitHub/autonomX/autonomx
```
Can we avoid hard-coding the path to `/Users/simondemeule/GitHub`, so that other developers can also use this script ?

aalex replied:
And also the path to the qt libs

aalex commented on runmactest:
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+PATH_TO_RELEASE_DOT_APP=/Users/simondemeule/GitHub/build-autonomx-Desktop_Qt_5_15_0_clang_64bit-Release/autonomx/autonomx.app
```
Can we avoid hard-coding the path to `/Users/simondemeule/GitHub`, so that other developers can also use this script ?

aalex reviewed (changes requested):
Can we avoid hard-coding the path to `/Users/simondemeule/GitHub`, so that other developers can also use this script ?

aalex commented:
- create a folder called scripts
- move the scripts this PR adds in that folder. Make sure they have a .sh file extension and are executable
- remove all the hard-coded paths and allow the scripts to find those executables and libraries automatically.

aalex commented:
See https://doc.qt.io/qt-5/macos-deployment.html

aalex commented:
We'll do a short pair-programming session in not too long to try to sort that out.

aalex committed:
Scripts to bundle the macOS app

Update the documentation about running macdeployqt

Added macos release scripts

Move the scripts and modify them

Update the scripts

Update the doc

aalex reviewed (approved):
LGTM
